src/utilities.hpp

```diff
@@ -183,9 +183,19 @@ std::string ToString(T value);
 
 // =================================================================================================
 
-// Helper for the function "GetArgument"
+// Helper for the function "GetArgument" to convert the argument to the value type
 template <typename T>
-T ConvertArgument(const char* value);
+T ConvertArgument(const char* value) {
+  return static_cast<T>(std::stoi(value));
+}
+
+// Variant of "ConvertArgument" with default values
+template <typename T>
+T ConvertArgument(const char* value, T default_value) {
+
+  if (value) { return ConvertArgument<T>(value); }
+  return default_value;
+}
```

Comment:
See above: can we move this to the .cpp file? And then add this to explicitly instantiate it for `size_t` (which is what you need for the platform/device settings - see below):
```
template size_t ConvertArgument(const char* value, size_t default_value);
```